test/unittests/mongoBackend/mongoContextProvidersQuery_test.cpp

```diff
@@ -625,11 +668,14 @@ TEST(mongoContextProvidersQueryRequest, noPatternAttrOneSingle)
 *
 * noPatternAttrOneMulti -
 *
-* Discover:  E1 - A1
-* Result:    E1 - A1 - http://cr1.com
-*            E1 - A1 - http://cr2.com
+* Query:  E1 - A1
+* Result: E1 - A1 - http://cr1.com
+*         E1 - A1 - http://cr2.com
 *
-* This test also checks that discovering for type (E1) doesn't match with no-typed
+* In this case, htpt://cr2.com also matches, but current implementation of mongoBackend
+* returns only one of them (which one is not specified).
```

Comment:
It is not easy... the logic in mongoBackend will change again.

However, don't worry about different machines. Although we only return one, it is deterministic (it is based on registrationQuery(), which returns results ordered by registration id).